grafana/provider.go

```diff
@@ -58,10 +68,46 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 	} else {
 		cfg.APIKey = auth[0]
 	}
+
+	headersObj := d.Get("headers").(map[string]interface{})
+	if headersObj != nil && len(headersObj) == 0 {
+		// Workaround for a bug when DefaultFunc returns a TypeMap
+		headersObjAbs, _ := JsonEnvDefaultFunc("GRAFANA_HTTP_HEADERS", nil)()
+		headersObj = headersObjAbs.(map[string]interface{})
+	}
+	if headersObj != nil {
+		// Convert headers from map[string]interface{} to map[string]string
+		headers := make(map[string]string)
+		for k, v := range headersObj {
+			switch v := v.(type) {
+			case string:
+				headers[k] = v
+			}
```

Comment:
For a single case type conversion, it's better to use a [type assertion](https://go.dev/ref/spec#Type_assertions) instead:
```suggestion
			if v, ok := v.(string); ok {
				headers[k] = v
			}
```

Reply:
Simplified.